File: Backend/GesthumServer/Models/Application.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace GesthumServer.Models
{
    public class Application
    {
        public int Id { get; set; }
        [ForeignKey("ResumeId")]
        public int ResumeId { get; set; }
        [ForeignKey("VacantId")]
        public int VacantId { get; set; }
        public DateTime ApplicationDate { get; set; } = DateTime.Now;
        public string Status { get; set; } = "Pending";
        public Resume Resume { get; set; }
        public Vacancy Vacant { get; set; }
    }
}

File: Frontend/GesthumClient/src/core/entities/Application.ts
export interface Application {
    id: number;
    // Identificador del resume (backend usa ResumeId)
    resumeId?: number;
    // Alias/compatibilidad: algunos lugares usan vacancyId, el backend expone VacantId
    vacancyId: number;
    vacantId?: number;
    // Compatibilidad histórica (si existiera)
    employeeId?: string;

    // Fecha en ISO (serializada desde DateTime del backend)
    applicationDate: string;

    // Estado (permitimos string para ser flexible con posibles nuevos valores)
    status: 'Pending' | 'Passed' | 'Failed' | string;

    // Datos de la vacante incluidos en el DTO del backend
    vacantTitle?: string;
    vacantLocation?: string;
    vacantPostedDate?: string;
    vacantCloseDate?: string;
    vacantState?: boolean;
}